tests/header_dependencies/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+#
+# Ensure all headers include all dependencies
+#
+
+set(IGNORED_HEADERS osvr/Util/MSStdIntC.h)
+
+set(COMMON_TEST_LIBRARIES jsoncpp_lib eigen-headers vendored-vrpn boost_system)
+
+add_custom_target(header_dependencies)
+
+file(GLOB_RECURSE headers RELATIVE "${HEADER_BASE}" ${HEADER_BASE}/*.h)
+foreach(HEADER ${headers})
+    list (FIND IGNORED_HEADERS "${HEADER}" _index)
+    if (${_index} GREATER -1)
+        continue()
+    endif()
+
+    string(REPLACE ".h" "" basename "${HEADER}")
+    string(REGEX REPLACE "[\\./]" "_" basename "${basename}")
```

Comment:
This step might be made easier/clearer/better by `string(MAKE_C_IDENTIFIER`, maybe? http://www.cmake.org/cmake/help/v3.0/command/string.html